### PR TITLE
build 1.5.5

### PR DESCRIPTION
**Pull request by megies**
Didn't check if dependencies have changed.. they're not listed in setup.py and I didn't bother to look through the whole code base.

Also, since github is not providing SHA sums for release tarballs, that's what I got from a manually downloaded tarball.. which kind of mocks out the whole concept.

### CODE REVIEW

**Comment by jakirkham** [comment]
Looks like it needs a [re-render]( https://github.com/conda-forge/conda-smithy#re-rendering-an-existing-feedstock ) as well.

**Comment by jakirkham** [comment]
This is a bit old, but here is what I found in [anaconda-recipes]( https://github.com/ContinuumIO/anaconda-recipes/blob/master/constructor/meta.yaml ). Seems the requirements have been changed.

**Comment by megies** [comment]
Ran into an error during re-rendering.. but it did *something*..

```
$ conda smithy rerender
No circle token.  Create a token at https://circleci.com/account/api and
put it in ~/.conda-smithy/circle.token
No appveyor token. Create a token at https://ci.appveyor.com/api-token and
Put one in ~/.conda-smithy/appveyor.token
No anaconda token. Create a token via
  anaconda auth --create --name conda-smithy --scopes "repos conda api"
and put it in ~/.conda-smithy/anaconda.token
Fetching package metadata .............
Traceback (most recent call last):
  File "/home/megies/anaconda/bin/conda-smithy", line 6, in <module>
    sys.exit(conda_smithy.cli.main())
  File "/home/megies/anaconda/lib/python2.7/site-packages/conda_smithy/cli.py", line 283, in main
    args.subcommand_func(args)
  File "/home/megies/anaconda/lib/python2.7/site-packages/conda_smithy/cli.py", line 188, in __call__
    configure_feedstock.main(args.feedstock_directory)
  File "/home/megies/anaconda/lib/python2.7/site-packages/conda_smithy/configure_feedstock.py", line 709, in main
    render_circle(env, config, forge_dir)
  File "/home/megies/anaconda/lib/python2.7/site-packages/conda_smithy/configure_feedstock.py", line 77, in render_circle
    forge_config.get('channels', {}).get('sources', tuple())
  File "/home/megies/anaconda/lib/python2.7/site-packages/conda_smithy/configure_feedstock.py", line 628, in compute_build_matrix
    mtx = special_case_version_matrix(meta, index)
  File "/home/megies/anaconda/lib/python2.7/site-packages/conda_build_all/version_matrix.py", line 223, in special_case_version_matrix
    py_vn = minor_vn(index[python_pkg.fn]['version'])
KeyError: u'python-3.3.0-pro0.tar.bz2'
```

**Comment by jakirkham** [comment]
Have you upgraded `conda-build-all`?

**Comment by mbargull** [comment]
Note that `1.6.0` is not a tag on the `master` branch but on another branch, associated with a pull request. `1.5.5` is the latest release from `master`.
(Also, I'd add `pyyaml` as a `run` requirement -- see the (declined) PR https://github.com/ContinuumIO/anaconda-recipes/pull/104)

**Comment by jakirkham** [comment]
I agree with your adjustments, @mbargull. Would you like to pick up this PR?

**Comment by megies** [comment]
> Have you upgraded conda-build-all?

Upgraded now (from 1.0.3 to 1.0.4), re-rendering now doesnt show an error (I did it again).

Stepped back to 1.5.5 and added pyyaml as dep.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -21,15 +21,16 @@ requirements:
...
-    - nsis          # [win]
+    - nsis >=3.01   # [win]
 
 test:
   imports:
     - constructor
-
   commands:
-    - constructor --help
+    - constructor --test
```
Comment (jakirkham) [inline recipe/meta.yaml]: Why not `--help`?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -21,15 +21,16 @@ requirements:
   run:
     - python
     - libconda
+    - pyyaml
+    - pycosat >=0.6.1
     - pillow >=3.1  # [win]
-    - nsis          # [win]
+    - nsis >=3.01   # [win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Guess this means we need to update `nsis`.

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipe/meta.yaml]: Guess [`AnacondaRecipes` does the same thing]( https://github.com/AnacondaRecipes/constructor-recipe/blob/85e1945d32f349d9b3c10642527a4ec71e0b3c48/recipe/meta.yaml#L25 ). Still curious as to why you made the switch, but it is non-blocking.

**Comment by jakirkham** [comment]
Thanks @megies. Sorry for the long delay.

File: recipe/meta.yaml (hunk 2 above)
Comment (jakirkham) [reply recipe/meta.yaml]: Updating in PR ( https://github.com/conda-forge/nsis-feedstock/pull/7 ).

File: recipe/meta.yaml (hunk 1 above)
Comment (megies) [reply recipe/meta.yaml]: Well, I guess you can blame it on my general packaging paranoia. ;-)
But I feel that even though the release itself should be properly tested, why not run tests if possible during packaging as well? There's always something can go wrong during packaging, even if the release is looking good.

**Pull request merged**